### PR TITLE
Java: Seed & indexable test

### LINKED ISSUE #388: Java: Sync callback
Im trying to interact with an MQTT stream from the rust side, but it only accepts `<Send + Sync + 'static>`

When trying to add the trait to my callback; like so:
`fn MqttTopicManager::subscribe(&self, cb: Box<dyn MqttListener + Send + Sync + 'static>) -> Result<()>;`

I receive the following error below.

Is this at all possible or do I need to find another way to do this?
```
rror[E0277]: `*mut *const jni_c_header::JNIInvokeInterface_` cannot be shared between threads safely
    --> /home/java/target/debug/build/client-db63cdf19f965487/out/java_glue.rs:5948:9
     |
5948 |         Box::new(cb)
     |         ^^^^^^^^^^^^ `*mut *const jni_c_header::JNIInvokeInterface_` cannot be shared between threads safely
     |
     = help: within `JavaCallback`, the trait `Sync` is not implemented for `*mut *const jni_c_header::JNIInvokeInterface_`
     = note: required because it appears within the type `JavaCallback`
     = note: required for the cast to the object type `dyn classes::mqtt::MqttListener + Sync + std::marker::Send`

error[E0277]: `*mut jni_c_header::_jobject` cannot be shared between threads safely
    --> /home/java/target/debug/build/client-db63cdf19f965487/out/java_glue.rs:5948:9
     |
5948 |         Box::new(cb)
     |         ^^^^^^^^^^^^ `*mut jni_c_header::_jobject` cannot be shared between threads safely
     |
     = help: within `JavaCallback`, the trait `Sync` is not implemented for `*mut jni_c_header::_jobject`
     = note: required because it appears within the type `JavaCallback`
     = note: required for the cast to the object type `dyn classes::mqtt::MqttListener + Sync + std::marker::Send`

error[E0277]: `*mut jni_c_header::_jmethodID` cannot be shared between threads safely
    --> /home/java/target/debug/build/client-db63cdf19f965487/out/java_glue.rs:5948:9
     |
5948 |         Box::new(cb)
     |         ^^^^^^^^^^^^ `*mut jni_c_header::_jmethodID` cannot be shared between threads safely
     |
     = help: the trait `Sync` is not implemented for `*mut jni_c_header::_jmethodID`
     = note: required because of the requirements on the impl of `Sync` for `Unique<*mut jni_c_header::_jmethodID>`
     = note: required because it appears within the type `alloc::raw_vec::RawVec<*mut jni_c_header::_jmethodID>`
     = note: required because it appears within the type `Vec<*mut jni_c_header::_jmethodID>`
     = note: required because it appears within the type `JavaCallback`
     = note: required for the cast to the object type `dyn classes::mqtt::MqttListener + Sync + std::marker::Send`
```

### LINKED ISSUE #790: [Task]: Add tests for verifying correct address derivation from seed
## Description

We lack tests that verify that address derivation is at par across implementations and bindings i.e., rust implementation and (java, nodejs, python) bindings. We should add them to be sure every implementation has correct address derivation. 

## Motivation

Extensive tests like these are needed to be confident about deterministic address derivation across all implementations. 

## Requirements

For a set of mnemonics, ensure they lead to deterministic addresses for different indexes. 

- Python — @bingyanglin;
- Java — @kwek20;
- Rust — @Thoralf-M;
- NodeJS — @laumair;
- Wasm — @laumair 

The mnemonics / seeds and expected addresses should be shared between all tests. 

**Tests should cover both derivations from mnemonics and also from hex seeds.**

## Open questions (optional)

Happy to hear if there are any other similar test scenarios that should be covered. 

## Are you planning to do it yourself in a pull request?

Yes — for nodejs bindings.

### PR DESCRIPTION
**Pull request by kwek20**
# Description of change

- Fixed the pom example to run from root or child
- Added tests for seed usage
- Added tests for indexable usage
- Made Payloads and Message serialisable
- Fixed MQTT(temp fix: https://github.com/Dushistov/flapigen-rs/issues/388)

## Links to any relevant issues

Fixes #790

## Type of change
- Bug fix (a non-breaking change which fixes an issue)
- Documentation Fix

## How the change has been tested

Ran the new tests, also verified with a non java made indexable

## Change checklist

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by thibault-martinez** [comment]
Is this ready to be reviewed/merged ?

**Comment by kwek20** [comment]
@thibault-martinez Yessir!

File: bindings/java/examples/java-app/src/test/java/org/iota/client/example/LibraryTest.java [hunk 1]
```diff
@@ -15,51 +18,124 @@ public class LibraryTest {
...
+        // get message data
+        Message message1Remote = iota.getMessage().data(fetched_message_ids[0]);
+        assertTrue(message1Remote.payload().isPresent());
+        verifyPayload(message1Remote.payload().get().asIndexation(), index, data);
+
+        // Make index from raw index payload
+        Message message2 = iota.message().finishIndex(IndexationPayload.fromStrings(index, data));
+        // Fetch message Ids
+        fetched_message_ids = iota.getMessage().indexString(index);
+        assertTrue(fetched_message_ids.length == 2);
```
Comment (Thoralf-M) [inline bindings/java/examples/java-app/src/test/java/org/iota/client/example/LibraryTest.java:81]: The amount of messages could change at any time if more messages get sent with this index (just running the test multiple times)

Comment (kwek20) [reply bindings/java/examples/java-app/src/test/java/org/iota/client/example/LibraryTest.java:81]: thats why the index has a date in it. unless you run it twice in a second or 2 people run it separately simultaneously :)

Comment (kwek20) [reply bindings/java/examples/java-app/src/test/java/org/iota/client/example/LibraryTest.java:81]: AH i did notice the order isnt deterministic, so i removed the ID equals check

**Comment by kwek20** [comment]
@Thoralf-M

**Review by Thoralf-M — APPROVED** [review]

**Comment by kwek20** [comment]
@thibault-martinez we can merge!

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**